javascript/app.js

```diff
@@ -0,0 +1,336 @@
+	// break name into single letters
+	let breakApart = [...ninja[i]];
+	// test break
+	// console.log(breakApart);
+	// look at name's single letters 
+	for (var j = 0; j < breakApart.length; j++) {
+		// every other letter
+		if (j % 2 === 0) {
+			// make uppercase
+			breakApart[j] = breakApart[j].toUpperCase();
+		}
+	}
+	// test uppercase
+	console.log(breakApart.join(''));
+}
+
+
+// RETURN OF THE CLOSETS
+const kristynsCloset = [
+  "left shoe",
+  "cowboy boots",
+  "right sock",
+  "GA hoodie",
+  "green pants",
+  "yellow knit hat",
+  "marshmallow peeps"
+];
+// Thom's closet is more complicated. Check out this nested data structure!!
+const thomsCloset = [
+  [
+    // These are Thom's shirts
+    "grey button-up",
+    "dark grey button-up",
+    "light blue button-up",
+    "blue button-up",
+  ],[
+    // These are Thom's pants
+    "grey jeans",
+    "jeans",
+    "PJs"
+  ],[
+    // Thom's accessories
+    "wool mittens",
+    "wool scarf",
+    "raybans"
+  ]
+];
+
+// Alien Attire
+// 1. Kristyn's left shoe has traveled through time and space and turned up in Thom's accessories drawer! Remove Kristyn's shoe from the array and save it to the variable kristynsShoe. Use that variable to add Kristyn's lost shoe to Thom's accessories array.
+
+// take the value (its an object)
+let kristynsShoe = kristynsCloset.splice(0, 1);
+// turn it into a string;
+kristynsShoe = kristynsShoe[0];
+// add to thom's accessories
+thomsCloset[2].push(kristynsShoe);
+// print to make sure
+console.log(kristynsCloset, thomsCloset);
+
+// Dress 'em Up
+// 1. Modify your code to put together 3 separate outfits for Kristyn and Thom. Put the output in a sentence to tell us what we'll be wearing. Mix and match!
+
+// KRISTYN
+// outfit combo declaration
+let outfitComboKristyn = [];
+// three possible outfits
+for (let h = 0; h < 3; h++){
+	// pick two pieces for one outfit
+	for (let i = 0; i < 2; i++) {
+		// at random
+		let randomValue = Math.floor(Math.random() * (kristynsCloset.length - outfitComboKristyn.length));
+		// test randomValue
+		console.log("Random value is " + randomValue);
+		// remove item from closet and put it into outfitComboKristyn
+		outfitComboKristyn.push(kristynsCloset.splice(randomValue, 1));
+	}
+	// to keep track of array
+	// console.log(outfitComboKristyn);
+	console.log(`Today Kristyn might wear ${outfitComboKristyn[0]} and ${outfitComboKristyn[1]}.`);
+	outfitComboKristyn = [];
+}
+
+// THOM
+// outfit combo declaration
+let outfitComboThom = [];
+// three possible outfits
+for (let h = 0; h < 3; h++){
+	// pick two pieces for one outfit
+	for (let i = 0; i < 2; i++) {
+		// random number to choose which drawer to take from
+		let randomValue = Math.floor(Math.random() * 2);
+		// test randomValue
+		console.log("Random value is " + randomValue);
+		// random number to choose which ITEM to take in drawer
+		let randomValue2 = Math.floor(Math.random() * (thomsCloset[randomValue].length - 1));
```

Comment:
subtracting 1 here is not helping either.  you sometimes get negative array indices.....

Reply:
also, to help yourself debug this, try using more semantic names.  instead of randomValue, call it `randomDrawerIndex`, or if you prefer `randDrawerIdx`.  instead of random value 2, use randomItemIndex or some such thing.